crates/polars-core/src/chunked_array/cast.rs

```diff
@@ -349,6 +349,12 @@ impl ChunkCast for StringChunked {
                 };
                 out
             },
+            #[cfg(feature = "dtype-time")]
+            DataType::Time => {
+                let result = cast_chunks(&self.chunks, data_type, options)?;
+                let out = Series::try_from((self.name(), result))?;
+                Ok(out)
```

Comment:
You're using `?` to early return an Err if it was produced, and then turning around and immediately returning an `Ok` of that expression.

`Ok(thing?)` === `thing`.

I'd suggest something like
``` rust
            DataType::Time => {
                let result = cast_chunks(&self.chunks, data_type, options)?;
                Series::try_from((self.name(), result))
            }
```